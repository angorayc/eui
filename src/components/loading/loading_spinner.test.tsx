import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import {
  EuiLoadingSpinner,
  SIZES,
  EuiLoadingSpinnerComponentType,
} from './loading_spinner';

describe('EuiLoadingSpinner', () => {
  test('is rendered', () => {
    const component = render(<EuiLoadingSpinner {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });

  describe('size', () => {
    SIZES.forEach(size => {
      test(`${size} is rendered`, () => {
        const component = render(<EuiLoadingSpinner size={size} />);

        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('component', () => {
    (['div', 'span', 'figure'] as EuiLoadingSpinnerComponentType[]).forEach(
      value => {
        test(`${value} is rendered`, () => {
          const component = render(<EuiLoadingSpinner component={value} />);

          expect(component).toMatchSnapshot();
        });
      }
    );

    ['h2'].forEach(value => {
      test(`${value} is not rendered`, () => {
        expect(() =>
          render(
            // intentionally passing an invalid value
            // @ts-ignore
            <EuiLoadingSpinner component={value} />
          )
        ).toThrow();
      });
    });
  });
});
