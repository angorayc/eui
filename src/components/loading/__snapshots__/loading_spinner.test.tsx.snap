// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiLoadingSpinner component div is rendered 1`] = `
<div
  class="euiLoadingSpinner euiLoadingSpinner--medium"
/>
`;

exports[`EuiLoadingSpinner component figure is rendered 1`] = `
<figure
  class="euiLoadingSpinner euiLoadingSpinner--medium"
/>
`;

exports[`EuiLoadingSpinner component span is rendered 1`] = `
<span
  class="euiLoadingSpinner euiLoadingSpinner--medium"
/>
`;

exports[`EuiLoadingSpinner is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiLoadingSpinner euiLoadingSpinner--medium testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;

exports[`EuiLoadingSpinner size l is rendered 1`] = `
<div
  class="euiLoadingSpinner euiLoadingSpinner--large"
/>
`;

exports[`EuiLoadingSpinner size m is rendered 1`] = `
<div
  class="euiLoadingSpinner euiLoadingSpinner--medium"
/>
`;

exports[`EuiLoadingSpinner size s is rendered 1`] = `
<div
  class="euiLoadingSpinner euiLoadingSpinner--small"
/>
`;

exports[`EuiLoadingSpinner size xl is rendered 1`] = `
<div
  class="euiLoadingSpinner euiLoadingSpinner--xLarge"
/>
`;
